benchexec/tools/mopsa.py

```diff
@@ -0,0 +1,55 @@
+# This file is part of BenchExec, a framework for reliable benchmarking:
+# https://github.com/sosy-lab/benchexec
+#
+# SPDX-FileCopyrightText: 2022 Raphaël Monat
+#
+# SPDX-License-Identifier: Apache-2.0
+
+import benchexec.tools.template
+import benchexec.result as result
+
+import re, json, subprocess
+
+
+class Tool(benchexec.tools.template.BaseTool2):
+    """
+    Tool info for Mopsa.
+    URL: https://gitlab.com/mopsa/mopsa-analyzer/
+    """
+
+    def executable(self, tool_locator):
+        return tool_locator.find_executable("mopsa-sv-comp", subdir="bin/")
+
+    def version(self, executable):
+        # the provided utility will fail due to return code 2 in Mopsa
```

Comment:
Maybe check for return codes here, too? I.e., provide a warning if the return code is not 2?

Furthermore, I would suggest to add error handling for cases like invalid JSON.

Reply:
You're right. I've changed the wrapper to make it compatible with `BaseTool2._version_from_tool`.